src/contract_leaderboard/mod.rs

```diff
@@ -0,0 +1,39 @@
+use crate::state::AppState;
+use axum::{extract::State, Json};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+#[derive(Serialize, Deserialize)]
+pub struct ContractEntry {
+    address: String,
+    base_fees: f64,
+    base_fees_usd: Option<f64>,
+}
+
+pub async fn get_contracts(State(state): State<Arc<AppState>>) -> Json<Vec<ContractEntry>> {
+    let db_pool = &state.db_pool;
+    let contracts: Vec<ContractEntry> = sqlx::query!(
+        "
+        SELECT
+            contract_address,
+            base_fee_sum,
+            base_fee_sum_usd
+        FROM
+            contract_base_fee_sums
+        WHERE base_fee_sum IS NOT NULL
+        ORDER BY base_fee_sum DESC
+        LIMIT 100;
+        "
+    )
+    .fetch_all(db_pool)
+    .await
+    .unwrap()
+    .into_iter()
+    .map(|row| ContractEntry {
+        address: row.contract_address,
```

Comment:
What other fields do we want to return here ? How about Metadata from the `contracts` table such as name, type, isBot etc. . Twitter data ? 

Are we sticking to only returning the base_fees (i.e. burned fees) for now or are we also planning to include the "tips" as well (which we don't calculate anywhere yet afaik) ?

Reply:
Let's start with just address and base fees. I anticipate name, image, and perhaps description are soon to follow.